lib/src/Uncertainty/Distribution/MethodOfMomentsFactory.cxx

```diff
@@ -0,0 +1,291 @@
+    , refMoments_(refMoments)
+    , distribution_(distribution)
+    , knownParameterValues_(knownParameterValues)
+    , knownParameterIndices_(knownParameterIndices)
+  {
+    distribution_.getImplementation()->setParallel(isParallel);
+
+    // build the unknown indices
+    const UnsignedInteger effectiveParameterSize = distribution.getParameter().getSize();
+    for (UnsignedInteger j = 0; j < effectiveParameterSize; ++ j)
+    {
+      if (!knownParameterIndices_.contains(j))
+        unknownParameterIndices_.add(j);
+    }
+  }
+
+  MethodOfMomentsEvaluation * clone() const
+  {
+    return new MethodOfMomentsEvaluation(*this);
+  }
+
+  UnsignedInteger getInputDimension() const
+  {
+    return distribution_.getParameterDimension() - knownParameterValues_.getSize();
+  }
+
+  UnsignedInteger getOutputDimension() const
+  {
+    return 1;
+  }
+
+  Description getInputDescription() const
+  {
+    return Description::BuildDefault(getInputDimension(), "theta");
+  }
+
+  Description getOutputDescription() const
+  {
+    return Description(1, "lh");
+  }
+
+  Description getDescription() const
+  {
+    Description description(getInputDescription());
+    description.add(getOutputDescription());
+    return description;
+  }
+
+  NumericalPoint operator() (const NumericalPoint & parameter) const
+  {
+    UnsignedInteger parameterDimension = distribution_.getParameterDimension();
+    NumericalPoint effectiveParameter(parameterDimension);
+    // set unknown values
+    UnsignedInteger unknownParameterSize = unknownParameterIndices_.getSize();
+    for (UnsignedInteger j = 0; j < unknownParameterSize; ++ j)
+    {
+      effectiveParameter[unknownParameterIndices_[j]] = parameter[j];
+    }
+    // set known values
+    UnsignedInteger knownParametersSize = knownParameterIndices_.getSize();
+    for (UnsignedInteger j = 0; j < knownParametersSize; ++ j)
+    {
+      effectiveParameter[knownParameterIndices_[j]] = knownParameterValues_[j];
+    }
+    Distribution distribution(distribution_);
+    try
+    {
+      distribution.setParameter(effectiveParameter);
+    }
+    catch (Exception &)
+    {
+      return NumericalPoint(1, SpecFunc::MaxNumericalScalar);
+    }
+
+    // compute moments of conditioned distribution
+    NumericalPoint moments(parameterDimension);
+    moments[0] = distribution.getMean()[0];
+    for (UnsignedInteger j = 1; j < parameterDimension; ++ j)
+    {
+      moments[j] = distribution.getCenteredMoment(j + 1)[0];
+    }
+
+    // compute sum of deltas between centered homogenized moments
+    NumericalScalar result = 0.0;
+    for (UnsignedInteger j = 0; j < parameterDimension; ++ j)
+    {
+      const NumericalScalar signref = refMoments_[j] > 0.0 ? 1.0 : -1.0;
+      const NumericalScalar sign = moments[j] > 0.0 ? 1.0 : -1.0;
+      result += std::abs(signref * std::pow(std::abs(refMoments_[j]), 1.0 / (j + 1.0)) - sign * std::pow(std::abs(moments[j]), 1.0 / (j + 1.0)));
+    }
```

Comment:
Here you have to compare the normalized moments, ie you have to compute:
[ |\mu_1-m_1|/\sigma ]^2 + \sum_{i=2}^K[ (\epsilon_i|\mu_i|^{1/i}-\eta_i|m_i|^{1/i})/\sigma ]^2
where \mu_1 is the mean of the distribution, m_1 the mean of the sample, \mu_i the centered moment of order i>=2 of the distribution, m_i the associated sample centered moment and \sigma is either the sample standard deviation or the distribution standard deviation. Note that you must use the same value to normalize the distribution moments and the sample moments. If you use the distribution standard deviation to normalize the distribution centered moments and the sample standard deviation to normalize the sample centered moments, the first term in the sum is equal to 0 and you loose one moment equation!